src/System.Net.Http/src/uap/System/Net/HttpClientHandler.cs

```diff
@@ -645,10 +650,35 @@ private void SetOperationStarted()
         {
             if (!_operationStarted)
             {
-                // Since this is the first operation, we set the proxy and server credentials on the
-                // WinRT filter based on the .NET handler's property settings.
+                // Since this is the first operation, we set all the necessary WinRT filter properties.
                 SetFilterProxyCredential();
                 SetFilterServerCredential();
+                if (_serverCertificateCustomValidationCallback != null)
+                {
+                    // The WinRT layer uses a different model for the certificate callback. The callback is
+                    // considered "extra" validation. We need to explicitly ignore errors so that the callback
+                    // will get called.
+                    //
+                    // In addition, the WinRT layer restricts some errors so that they cannot be ignored, such 
+                    // as "Revoked". This will result in behavior differences between UWP and other platforms.
+                    // The following errors cannot be ignored right now in the WinRT layer:
+                    //
+                    //     ChainValidationResult.BasicConstraintsError
+                    //     ChainValidationResult.InvalidCertificateAuthorityPolicy
+                    //     ChainValidationResult.InvalidSignature
+                    //     ChainValidationResult.OtherErrors
+                    //     ChainValidationResult.Revoked
+                    //     ChainValidationResult.UnknownCriticalExtension
+                    _rtFilter.IgnorableServerCertificateErrors.Add(RTChainValidationResult.Expired);
+                    _rtFilter.IgnorableServerCertificateErrors.Add(RTChainValidationResult.IncompleteChain);
+                    _rtFilter.IgnorableServerCertificateErrors.Add(RTChainValidationResult.InvalidName);
+                    _rtFilter.IgnorableServerCertificateErrors.Add(RTChainValidationResult.RevocationFailure);
+                    _rtFilter.IgnorableServerCertificateErrors.Add(RTChainValidationResult.RevocationInformationMissing);
+                    _rtFilter.IgnorableServerCertificateErrors.Add(RTChainValidationResult.Untrusted);
+                    _rtFilter.IgnorableServerCertificateErrors.Add(RTChainValidationResult.WrongUsage);
+                    _rtFilter.ServerCustomValidationRequested += RTServerCertificateCallback;
```

Comment:
> We can either throw an exception during the property setter or the callback

For reference, we have a similar issue with CurlHandler, and we throw the PNSE during the request/callback.  While in some situations we could detect this based purely on inspecting the libcurl version/build being used, in general we don't know whether a callback is supported until we try to register it with libcurl and it gives us an answer about whether it could hook it up.